src/api/migrations.ts

```diff
@@ -56,6 +59,49 @@ const cloneRepo = ({ config }: AppState) => async (
   return repoDir
 }
 
+const cloneRepoWithAbSync = (appState: AppState) => async (
+  repoId: string
+): Promise<string | undefined> => {
+  const repoDirs = await Promise.all(
+    appState.config.migrationOriginServers.map(url =>
+      cloneRepo(appState)(url, repoId)
+    )
+  )
```

Comment:
Use `Promise.allSettled` and throw on any errors other than repo not found error from cloneRepo